packages/offix-datastore/src/Model.ts

```diff
@@ -89,4 +133,34 @@ export class Model<T = unknown> {
                 listener(event);
             });
     }
+
+    private async subscribeForServerEvents(replicator: IReplicator) {
+        // TODO replicator.subscribe
+        // handle subscription events here
+    }
+
+    private async doDeltaSync(replicator: IReplicator, matcher: (d: T) => Predicate<T>, predicate?: Predicate<T>) {
+        // TODO limit the size of data returned
+        // TODO get lastSync for model for metadata store and pass to pullDelta
+        const data = await replicator.pullDelta(this.getStoreName(), "", predicate);
+
+        data
+            .filter((d: any) => (d._deleted))
+            .forEach((d: any) => { this.remove(matcher(d)); });
```

Comment:
I would like to know why we doing this? What is the use case.

Reply:
I don't understand what you mean here. Are you asking about the matcher?

Reply:
Why we remove `_deleted` elements?